### PR TITLE
update deps

### PR DESCRIPTION
**Pull request by BenTheElder**
with a bit of fiddling I've shrunk the root go.mod and updated dependencies

### CODE REVIEW

**Comment by BenTheElder** [comment]
steps:
- update direct dependencies (NOT client-go and k8s.io/api)
- drop all //indirect from go.mod
- add back k8s.io/api and k8s.io/apimachinery indirects
- hand edit client-go and k8s.io/ to updated matching versions
- result in this `go.mod`:
```
module sigs.k8s.io/kind

require (
	github.com/google/gofuzz v1.0.0
	github.com/google/uuid v1.1.1
	github.com/pkg/errors v0.8.0
	github.com/sirupsen/logrus v1.4.1
	github.com/spf13/cobra v0.0.3
	golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
	k8s.io/api kubernetes-1.14.1 // indirect
	k8s.io/apimachinery kubernetes-1.14.1
	k8s.io/client-go v11.0.0+incompatible
	sigs.k8s.io/kustomize v2.0.3+incompatible
)
```
- `make build` to add in updated indirect dependencies

File: go.mod [hunk 1]
```diff
@@ -1,36 +1,27 @@
...
+	github.com/gogo/protobuf v1.2.1 // indirect
+	github.com/google/gofuzz v1.0.0
 	github.com/google/uuid v1.1.1
-	github.com/googleapis/gnostic v0.2.0 // indirect
 	github.com/inconshreveable/mousetrap v1.0.0 // indirect
-	github.com/json-iterator/go v1.1.5 // indirect
-	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
-	github.com/modern-go/reflect2 v0.0.0-20180701023420-4b7aa43c6742 // indirect
-	github.com/onsi/ginkgo v1.7.0 // indirect
-	github.com/onsi/gomega v1.4.3 // indirect
```
Comment (BenTheElder) [inline go.mod]: i'm really not sure how these were still in the root go.mod (ginkgo, gomega), happy to have the dependencies reduced, it should make building faster :-)

**Comment by BenTheElder** [comment]
added a similar update for the tools module. same pinning method (1.14.1 kubernetes-wise)

File: hack/tools/go.sum [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
-k8s.io/gengo v0.0.0-20181113154421-fd15ee9cc2f7 h1:zjNgw2qqBQmKd0S59lGZBQqFxJqUZroVbDphfnVm5do=
-k8s.io/gengo v0.0.0-20181113154421-fd15ee9cc2f7/go.mod h1:ezvh/TsK7cY6rbqRK0oQQ8IAqLxYwwyPxAX1Pzy0ii0=
-k8s.io/klog v0.1.0 h1:I5HMfc/DtuVaGR1KPwUrTc476K8NCqNBldC7H4dYEzk=
-k8s.io/klog v0.1.0/go.mod h1:Gq+BEi5rUBO/HRz0bTSXDUcqjScdoY3a9IHpCEIOOfk=
+k8s.io/code-generator v0.0.0-20190311093542-50b561225d70 h1:lgPp615xLHxN84RBd+viA/oHzJfI0miFYFH4T9wpPQ4=
+k8s.io/code-generator v0.0.0-20190311093542-50b561225d70/go.mod h1:MYiN+ZJZ9HkETbgVZdWw2AsuAi9PZ4V80cwfuf2axe8=
+k8s.io/gengo v0.0.0-20190327210449-e17681d19d3a h1:QoHVuRquf80YZ+/bovwxoMO3Q/A3nt3yTgS0/0nejuk=
+k8s.io/gengo v0.0.0-20190327210449-e17681d19d3a/go.mod h1:ezvh/TsK7cY6rbqRK0oQQ8IAqLxYwwyPxAX1Pzy0ii0=
+k8s.io/klog v0.3.0 h1:0VPpR+sizsiivjIfIAQH/rl8tan6jvWkS7lU+0di3lE=
+k8s.io/klog v0.3.0/go.mod h1:Gq+BEi5rUBO/HRz0bTSXDUcqjScdoY3a9IHpCEIOOfk=
```
Comment (neolit123) [inline hack/tools/go.sum]: just wanted to note that the whole idea of suffixing a semver with `/go.mod` is crazy-pants syntax.

/lgtm

Comment (BenTheElder) [reply hack/tools/go.sum]: hehe

Comment (BenTheElder) [reply hack/tools/go.sum]: we may also want to prune these someday, working as intended is growing infinitely with all observed dependency hashes...

**Pull request merged**